### PR TITLE
Appstudio error handling

### PR DESCRIPTION
**Pull request by simonbaird**
JIRA: [HACBS-2357](https://issues.redhat.com/browse/HACBS-2357)

This is closely related to #836 but I wanted it to be reviewed separately.

### CODE REVIEW

**Comment by simonbaird** [comment]
Now #836 is merged, this is ready for review.

File: acceptance/examples/broken_input.json [hunk 1]
```diff
@@ -0,0 +1 @@
+{"fu":"bar}
```
Comment (simonbaird) [inline acceptance/examples/broken_input.json:1]: ...a challenge for delinters everywhere. :grin:

File: cmd/test/test.go [hunk 2]
```diff
@@ -224,6 +230,28 @@ func newTestCommand() *cobra.Command {
...
+// Special error handling for appstudio format only
+func appstudioErrorHandler(noFail bool, prefix string, err error) error {
+	// Output some json to stdout
+	applicationsnapshot.OutputAppstudioReport(applicationsnapshot.AppstudioReportForError())
+
+	// Beware we're effectively changing the meaning of the --no-fail flag here.
+	// Rather than being only about policy failures any more, we're extending
+	// it to also mean don't return a non-zero exit code for an error handled
+	// by this function.
+	if noFail {
```
Comment (zregvart) [inline cmd/test/test.go:242]: Is this something we want to replicate in other commands?

**Review by zregvart — APPROVED** [review]
LGTM

File: cmd/test/test.go (hunk 2 above)
Comment (simonbaird) [reply cmd/test/test.go:242]: Hmm, maybe. It's a similar to --strict used elsewhere.

**Comment by zregvart** [comment]
Merge and refactor as need be?

**Comment by simonbaird** [comment]
Let's merge.

**Comment by simonbaird** [comment]
There were a few minor tweaks needed to make it work after the rebase on #879.

**Pull request merged**